### ISSUE TITLE
Documentation in complete. 

### ISSUE THREAD

**Issue by mtekp** [actix/actix#172]
I am a scala developer with extensive use of akka. And hate its Any => Unit type for message handlers. I really liked rust for its safety and speed. I came across actix and wanted to learn it. But the main parts are execution context, scheduling, supervision  and many more which are undocumented and I just see [WIP]. Please give this priority as it will help new comers to understand how actix work.

**Comment by Firstyear** [comment]
I agree. I think soon if I have time I'll start to write these myself as I have been muddling my way through to understand it.

**Comment by najamelan** [comment]
Yes, please! Actix looks really great, but it's not straight forward...

**Comment by Freyskeyd** [comment]
Anyone working on some documentation improvement?

It's really hard to make things work without it.
Some keys features aren't documented at all.. like streams

**Comment by Firstyear** [comment]
@Freyskeyd I was, but I ended up distracted. There is a draft of the streams docs as a PR here: https://github.com/actix/book/pull/30

**Comment by Freyskeyd** [comment]
Thank's @Firstyear 

I was already using `add_stream` and `add_stream_message` but I was interested in `ActorStream` which is like an `ActorFuture` but for stream. I didn't find any usage anywhere nor `Mocker` which is a key feature for testing.

**Comment by apehub** [comment]
> Thank's @Firstyear
> 
> I was already using `add_stream` and `add_stream_message` but I was interested in `ActorStream` which is like an `ActorFuture` but for stream. I didn't find any usage anywhere nor `Mocker` which is a key feature for testing.
> 
> For `ActorStream` I was looking for an example of how it would work with a stream consuming a `VecDeque` internally, I want to send a message to a process and get a stream of those values.

@Freyskeyd 

I am  studying the official "chat" example, but I have some question about the "ctx.add_message_stream": 

When I run the server without change any code, the example works well: the "12345" port is listening and the "clients" can connect to the chat server and send messages.

Here are the origin codes:

https://github.com/actix/actix/blob/75d960270cae21cdd0c69f44641cb83e643e11b0/examples/chat/src/main.rs#L67-L70

But when i remove the "ctx.add_stream_message", the server can start normally, but the "12345" port is not listening and client cannot connect to the server.

Here are the codes I remove the "ctx.add_message_stream"

```rust
Server::create(|ctx| {
    // ctx.add_message_stream(
        listener.incoming()
            .map_err(|_| ())
            .map(|st| {
                let addr = st.peer_addr().unwrap();
                TcpConnect(st, addr)
            });
    //  );
        Server { chat: server }
    }
);
```

So, my question is: do you know what magic the "ctx.add_message_stream" did? And what the codes looks like if I donot use the "ctx.add_message_stream" ?

Thanks ~

**Comment by Freyskeyd** [comment]
Hello @apehub 

If you remove the `add_message_stream` the `listener` future will never be poll. So you don't have a server that are listening for connection.

I think it's something like that :)

**Comment by apehub** [comment]
> Hello @apehub
> 
> If you remove the `add_message_stream` the `listener` future will never be poll. So you don't have a server that are listening for connection.
> 
> I think it's something like that :)

Hello, @Freyskeyd , thanks for your comments :)

**Comment by jduan** [comment]
Agree. Having solid documentation is essential for attracting new users.

**Comment by naturallymitchell** [comment]
See also https://github.com/rust-lang/rfcs/issues/819
